### PR TITLE
Add iOS build instruction

### PR DESCRIPTION
**Pull request by gwang-msft**
**Description**: Add iOS build instruction

**Motivation and Context**
- We do not have any build instruction of iOS yet

### CODE REVIEW

**Review by gwang-msft — COMMENTED** [review]
Reverted an accidental commit to the wrong branch

File: BUILD.md [hunk 1]
```diff
@@ -1017,6 +1018,32 @@ Android NNAPI Execution Provider can be built using building commands in [Androi
...
+* General Info:
+   * iOS Architectures
+
+      | ios_sysroot     | arm64       | x86_64      |
+      |-----------------|:-----------:|:-----------:|
+      | iphoneos        | YES         | NO          |
+      | iphonesimulator | NO          | YES         |
+      armv7, armv7s and i386 architectures are not currently supported.
+   * apple_deploy_target
+
```
Comment (skottmckay) [inline BUILD.md:1036]: Link to apple doc describing what the valid values are would be good.

File: BUILD.md [hunk 2]
```diff
@@ -1017,6 +1018,32 @@ Android NNAPI Execution Provider can be built using building commands in [Androi
...
+* Xcode
+   * Install Xcode from https://developer.apple.com/xcode/
+* General Info:
+   * iOS Architectures
+
+      | ios_sysroot     | arm64       | x86_64      |
+      |-----------------|:-----------:|:-----------:|
+      | iphoneos        | YES         | NO          |
+      | iphonesimulator | NO          | YES         |
+      armv7, armv7s and i386 architectures are not currently supported.
```
Comment (skottmckay) [inline BUILD.md:1034]: Not clear to me how to interpret this but possibly as I know nothing about this setup. 

Is it that all mac's are x86_64 so you have to use the simulator if testing on a laptop/desktop and specify x86_64, and all iphones are arm64 so you have to build with arm64 to run on an actual device?

File: BUILD.md [hunk 3]
```diff
@@ -1017,6 +1018,32 @@ Android NNAPI Execution Provider can be built using building commands in [Androi
...
+   * apple_deploy_target
+
+      Specify the minimum version of the target platform (iOS) on which the target binaries are to be deployed.
+
+#### Build Instructions
+```
+./build.sh --config <Release|Debug|RelWithDebInfo|MinSizeRel> \
+           --ios --ios_sysroot <iphoneos|iphonesimulator> --osx_arch <arm64|x86_64> \
```
Comment (skottmckay) [inline BUILD.md:1042]: Mention build.sh is run from in the ORT repo root

File: BUILD.md (hunk 1 above)
Comment (gwang-msft) [reply BUILD.md:1031]: Unfortunately, Apple does not have any official document about this

**Review by skottmckay — APPROVED** [review]

**Pull request merged**